share/doc/wake/logging.md

```diff
@@ -0,0 +1,64 @@
+# Logging
+
+This article gives a high level overview of how wake logging works.
+Check out [print.wake](https://github.com/sifive/wake/blob/master/share/wake/lib/core/print.wake)
+for commented code.
+
+## How Logging works on Wake Execution commands
+
+Every `Job` will emit a stderr and a stdout.
+
+When writing your wake code, you can control where each goes with `setPlanStderr` and `setPlanStdout` before you `runJob`.
+We call this which `stream` each goes to (we often refer to a `stream` as a `logLevel`).
+
+Regardless of which `stream` you direct the `Job`'s `Stderr` and `Stdout`,
+`wake` will store all the text in its database,
+discarding the information about the `stream` it was destined for at the time it was stored.
+It will however know whether it came from the `Job`'s stderr/stdout.
+
+So what's the difference between the different streams?
+What you see on your console when you run wake.
+It controls *what* you see (depending on the command line flags) and what it looks like (what color it is).
+It also controls which *actual* stream it is going to coming out of `wake`
+(stderr or stdout, or other file descriptors 3-5).
```

Comment:
This sentence is a bit confusing because you are using the word "stream" differently than in the first sentence of the paragraph. It's almost tautological because you are effectively saying that the difference between the different "streams" is which "stream" they use.

Perhaps we just need to avoid using the term "stream" to refer to a wake `LogLevel`, since it makes it difficult for us to refer to the [Unix concept of the same name](https://en.wikipedia.org/wiki/Standard_streams).

Notably, a Unix stream doesn't dictate the color because color codes can be present within the stream to change the colors of specific characters displayed by a terminal, rather than the entire stream.

If we're going to bikeshed about naming here, I'd like to pitch renaming `LogLevel` to `Logger`, since I think that more closely matches how many other languages ([Java](https://docs.oracle.com/javase/7/docs/api/java/util/logging/Logger.html), [Python](https://docs.python.org/3/howto/logging.html)) call it, where the logging level is just a property of a logger.

Reply:
right, i found it really awkward to call it a "log level" as the sentences don't make sense.

But what i am trying to distinguish here is exactly what you helped rewrite in the first sentence -- each running job has its own output streams. Where you tell wake to send it via the `logLevel` mechanis, to one of *wake*'s output streams is different than those.

Reply:
used logger instead of stream. It reads much better now, thank you.